airbyte-config/config-models/src/main/java/io/airbyte/config/helpers/StateMessageHelper.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.config.helpers;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.StateType;
+import io.airbyte.config.StateWrapper;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import java.util.List;
+import java.util.Optional;
+
+public class StateMessageHelper {
+
+  public static class AirbyteStateMessageListTypeReference extends TypeReference<List<AirbyteStateMessage>> {}
+
+  /**
+   * This a takes a json blob state and tries return either a legacy state in the format of a json
+   * object or a state message with the new format which is a list of airbyte state message.
+   *
+   * @param state - a blob representing the state
+   * @return An optional state wrapper, if there is no state an empty optional will be returned
+   */
+  public static Optional<StateWrapper> getTypedState(final JsonNode state) {
+    if (state == null) {
+      return Optional.empty();
+    } else {
+      final List<AirbyteStateMessage> stateMessages;
+      try {
+        stateMessages = Jsons.object(state, new AirbyteStateMessageListTypeReference());
```

Comment:
This try/catch probably should be extracted to a method to make this a bit more readable.  Something like:

```
private List<AirbyteStateMessage> deserializeState(final JsonNode state) { ...
```